src/cefpython.pyx

```diff
@@ -1017,11 +1017,12 @@ def Shutdown():
     # and closes browser automatically if you give it some time.
     # If the time was not enough, then there is an emergency plan,
     # the code block further down that checks len(g_pyBrowsers).
-    for _ in range(20):
-        for __ in range(10):
-            with nogil:
-                CefDoMessageLoopWork()
-        time.sleep(0.01)
+    if not GetAppSetting("multi_threaded_message_loop"):
```

Comment:
Multi threaded message loop is a Windows-only feature, so on other systems when this setting is set to True it should be ignored. CefDoMessageLoopWork should still be called on Linux/Mac. You should introduce a local variable `multi_threaded_message_loop_windows` and use it instead.